Framework/Source/Events/CDESaveMonitor.m

```diff
@@ -197,6 +202,12 @@ - (void)asynchronouslyStoreChangesForContext:(NSManagedObjectContext *)context c
     NSDictionary *deleteData = [eventBuilder changesDataForDeletedObjects:deletedObjects inManagedObjectContext:context];
     [changedValuesByContext removeObjectForKey:context];
 
+    if(((NSArray*)insertData[@"changeArrays"]).count + ((NSArray*)updateData[@"objectIDs"]).count + ((NSArray*)deleteData[@"orderedObjectIDs"]).count == 0)
```

Comment:
Yeah, it looks like the event is created just under that, so bailing there should be fine.